### PR TITLE
SOLR-15919 Another pass as moving File to Path

### PR DESCRIPTION
**Pull request by madrob**
https://issues.apache.org/jira/browse/SOLR-15919

I'm happy to create a new JIRA if people prefer or we can lump it in with the previous one.

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/update/UpdateLog.java [hunk 1]
```diff
@@ -1328,25 +1325,27 @@ public void copyOverOldUpdates(long commitVersion, TransactionLog oldTlog) {
...
-      for (String oldBufferLogName : oldBufferTlog) {
-        deleteFile(new File(tlogDir, oldBufferLogName));
-      }
+    try (Stream<Path> tlogs = Files.walk(tlogDir, 1)) {
+      final String prefix = BUFFER_TLOG_NAME + '.';
+      tlogs.filter(Files::isRegularFile)
+          .filter(path -> path.getFileName().toString().startsWith(prefix))
+          .forEach(UpdateLog::deleteFile);
+    } catch (IOException e) {
+      log.warn("Could not clean up buffered transaction logs in {}", tlogDir, e);
```
Comment (janhoy) [inline solr/core/src/java/org/apache/solr/update/UpdateLog.java:1340]: Nice that you also add more informative exception messages

File: solr/core/src/java/org/apache/solr/update/UpdateLog.java [hunk 2]
```diff
@@ -2133,18 +2132,18 @@ public static AddUpdateCommand convertTlogEntryToAddUpdateCommand(SolrQueryReque
...
+      Files.deleteIfExists(file);
       success = true;
     } catch (Exception e) {
       log.error("Error deleting file: {}", file, e);
     }
 
     if (!success) {
       try {
-        file.deleteOnExit();
+        file.toFile().deleteOnExit();
```
Comment (janhoy) [inline solr/core/src/java/org/apache/solr/update/UpdateLog.java:2146]: Just looks a bit weird with `file.toFile()`, but probably ok. Have not checked if there is a deleteOnExit API on Files..

**Review by janhoy — COMMENTED** [review]
Browsed through and all looks safe. One TODO to handle.

File: solr/core/src/java/org/apache/solr/update/UpdateLog.java (hunk 2 above)
Comment (madrob) [reply solr/core/src/java/org/apache/solr/update/UpdateLog.java:2146]: There's not a deleteOnExit API for NIO. The options are to either convert to a File or register our own shutdown hooks (which is what this does behind the scenes). I think this code exists specifically for Windows running with an AV (or the test FileSystem that simulates this) so I'm not sure what the other replacement options are.

**Pull request merged**